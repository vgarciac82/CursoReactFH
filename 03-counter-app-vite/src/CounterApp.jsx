/*RAFC*/
import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value = -1 }) => {
  const [counter, setCounter] = useState(value);
  const handleAdd = () => {
    setCounter(counter + 1);
  };

  const handleSub = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };
  return (
    <>
      <h1>Counter APP</h1>
      <h2>{counter}</h2>

      <button onClick={handleAdd}> +1 </button>
      <button onClick={handleSub}> -1 </button>
      <button onClick={handleReset}> Reset </button>
    </>
  );
};

CounterApp.proppTypes = {
  value: PropTypes.number,
};
