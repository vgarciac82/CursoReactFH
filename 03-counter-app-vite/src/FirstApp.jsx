import PropTypes from "prop-types";

const FirstApp = ({ title, subtitle = "REACT DEVELOPER" }) => {
  const getResult = (a, b) => {
    return a + b;
  };

  return (
    <>
      
        <h1><div data-testid="test-title">{title}</div></h1>
      
      <h2>{subtitle}</h2>
      <p>My first React App</p>
    </>
  );
};

export default FirstApp;

FirstApp.proppTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  subtitle: "BEST REACT DEVELOPER",
};
