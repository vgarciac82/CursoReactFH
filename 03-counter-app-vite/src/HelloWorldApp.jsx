import React from 'react'

export const HelloWorldApp = () => {
  return (
    <div>Hola Mundo!!</div>
  )
}
