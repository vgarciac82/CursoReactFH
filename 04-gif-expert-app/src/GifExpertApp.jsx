import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["one punch", "Dragon Ball"]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GIF Expert App</h1>

      <AddCategory
        onNewCategory={(event) => onAddCategory(event)}
      ></AddCategory>

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};

export default GifExpertApp;