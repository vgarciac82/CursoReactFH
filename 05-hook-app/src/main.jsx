import React from "react";
import ReactDOM from "react-dom/client"; 
import { TodoApp } from "./08-useReducer/TodoApp";

 
import "./index.css";

import './08-useReducer/intro-reducer'
/*
ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <TodoApp />
  </React.StrictMode>
);
*/