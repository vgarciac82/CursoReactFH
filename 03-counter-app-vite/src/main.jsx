
import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import  FirstApp  from './FirstApp';

import './style.css';

ReactDOM.createRoot( document.getElementById('root') ).render( 
    <React.StrictMode>
        { /*<CounterApp value={ 100 }></CounterApp> */ }
        
        <FirstApp title="Hola Mundo"></FirstApp>
    </React.StrictMode>
 );