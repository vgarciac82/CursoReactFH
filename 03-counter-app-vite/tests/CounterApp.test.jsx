import { fireEvent, getByRole, render } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe("Pruebas al counter App", () => {
  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<CounterApp />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar el valor inicial de 100", () => {
    const val = 100;
    const { container, getByText } = render(<CounterApp value={val} />);
    expect(getByText(val)).toBeTruthy();
  });

  test("Debe incrementar con el boton de +1", () => {
    const val = 100;
    const { container, getByText } = render(<CounterApp value={val} />);

    fireEvent.click(getByText("+1"));

    expect(getByText("101")).toBeTruthy();
  });

  test("Debe decrementar con el boton de -1", () => {
    const val = 100;
    const { container, getByText } = render(<CounterApp value={val} />);

    fireEvent.click(getByText("-1"));

    expect(getByText("99")).toBeTruthy();
  });

  test("Debe funcionar   el boton de Reset", () => {
    const val = 100;
    const { container, getByText, getByRole } = render(<CounterApp value={val} />);

    fireEvent.click(getByText("-1"));
    fireEvent.click(getByText("-1"));
    fireEvent.click(getByText("-1"));

    //fireEvent.click(getByText("Reset"));
    fireEvent.click(getByRole('button', { name: 'btn-reset' }));

    expect(getByText("100")).toBeTruthy();
  });
  
});
