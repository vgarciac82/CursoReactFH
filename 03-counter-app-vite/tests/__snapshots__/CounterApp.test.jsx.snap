// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al counter App Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter APP
  </h1>
  <h2>
    100
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
